### PR TITLE
Clean up all selection limit enforcement

### LINKED ISSUE #4197: DFV: Clean up all selection limit enforcement
PR: #4218 

[Edit: This has become a generic "sweep up selection limits" issue]
- [x] I don't think there's a way to select _no_ items once there has been a selection (er, whoops) [ctrl-click here... it's just whatever standard multiselect behavior is for the platform]
- [x] We need a plan for how to handle consecutive selections that go over the limit (click-select).  This may need some discussion
- [x] Disable "add new" whenever selection limit is reached
- [x] Disable checkboxes to enforce limit, using the same approach as multiselect
- [x] Make sure all views still respect unlimited selection (limit = 0)

### PR DESCRIPTION
**Pull request by pglewis**
Fixes #4197

### CODE REVIEW

File: ui/js/pods-dfv/_src/pick/views/select-view.js [hunk 1]
```diff
@@ -145,19 +145,57 @@ export const SelectView = Marionette.CollectionView.extend( {
...
+		const view_name = this.fieldConfig.view_name;
+		const format_type = this.fieldConfig.pick_format_type;
+		const limit = this.fieldConfig.pick_limit;
+		let numSelected;
+
 		this.collection.setSelected( this.$el.val() );
+		numSelected = this.collection.filterBySelected().length;
+
+		if ( 'select' === view_name && 'multi' === format_type ) {
+			if ( numSelected >= limit ) {
```
Comment (sc0ttkclark) [inline ui/js/pods-dfv/_src/pick/views/select-view.js]: This should check that limit itself is `0 < limit` and please follow the `1 < 2` order of vars when comparing. So it'd be `limit <= numSelected`.

If limit is not above 0, then it shouldn't run enableAll() either.

File: ui/js/pods-dfv/_src/pick/views/select-view.js [hunk 2]
```diff
@@ -145,19 +145,57 @@ export const SelectView = Marionette.CollectionView.extend( {
...
+				this.enableAll();
+			}
+		}
+	},
+
+	/**
+	 *
+	 */
+	disableUnselected: function () {
+		this.$el.find( 'option:not(:selected)' ).prop( 'disabled', true );
```
Comment (sc0ttkclark) [inline ui/js/pods-dfv/_src/pick/views/select-view.js]: How does this work with Add New? When you add a new record, or edit, and it updates the item in the select list, does this not "stick"?

File: ui/js/pods-dfv/_src/pick/views/select-view.js (hunk 1 above)
Comment (pglewis) [inline ui/js/pods-dfv/_src/pick/views/select-view.js]: Working on a cleaner solution to that very thing on the checkbox side.

**Comment by pglewis** [comment]
Creating a task list in the first post

File: ui/js/pods-dfv/_src/pick/views/checkbox-view.js [hunk 3]
```diff
@@ -46,14 +46,44 @@ export const CheckboxView = PodsFieldListView.extend( {
...
-		const numSelected = this.collection.filterBySelected().length;
+		let limit, numSelected;
+
+		limit = +fieldConfig.pick_limit;  // Unary plus will implicitly cast to number
+		numSelected = this.collection.filterBySelected().length;
 
-		// Enforce selection limit, ignoring de-selection.  Note that 'selected' is the value before the change here
-		if ( numSelected < limit || childView.model.get( 'selected' ) ) {
-			childView.model.toggleSelected();
+		if ( 0 === limit || numSelected < limit ) {
```
Comment (sc0ttkclark) [inline ui/js/pods-dfv/_src/pick/views/checkbox-view.js]: Should `0 === limit` still trigger prop() changes? Should it return null here and check for null where validateSelectionLimit is used?

Comment (pglewis) [inline ui/js/pods-dfv/_src/pick/views/checkbox-view.js]: That's premature optimization IMO

**Comment by pglewis** [comment]
Not 100% ready for review/merge just yet.  A couple UI features left in the issue's tasklist.

**Pull request merged**